Shop_Unity_base/Assets/Scripts/Shop/Utility/Random.cs

```diff
@@ -0,0 +1,17 @@
+﻿namespace Utility
+{
+    public static class Random
+    {
+        private static System.Random instance;
+        public static System.Random Instance
+        {
+            get
+            {
+                if (instance == null)
+                    instance = new System.Random();
+
+                return instance;
+            }
+        }
```

Comment:
\>:)
> "Friends don't let friends do Singletons." — Rob Nystrom

Nystrom suggests a Service Locator pattern instead of this.

Reply:
I just did it to show I can make one lol.
Can probably replace with a static call just as easy.